pkg/limiters/global_limiter.go

```diff
@@ -0,0 +1,101 @@
+package limiters
+
+import (
+	"github.com/lwolf/konsumerator/pkg/helpers/tests"
+
+	"github.com/go-logr/logr"
+	konsumeratorv1alpha1 "github.com/lwolf/konsumerator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type GlobalLimiter struct {
+	availCPU *resource.Quantity
+	availMem *resource.Quantity
+	log      logr.Logger
+}
+
+func NewGlobalLimiter(policy *konsumeratorv1alpha1.ResourcePolicy, used *corev1.ResourceList, log logr.Logger) *GlobalLimiter {
+	l := &GlobalLimiter{log: log}
+	if policy == nil || policy.GlobalPolicy == nil {
+		// no limits were set
+		return l
+	}
+	if used == nil {
+		used = tests.NewResourceList("0", "0")
```

Comment:
it doesn't seem right to import from tests here. If this function is now not tests-specific, let's move it to generic helpers